src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/integTests/tests/jvmsizing/HeapSizeIncreaseHighThresholdTest.java

```diff
@@ -0,0 +1,282 @@
+                )
+            }
+        ),
+        @ATable(
+            hostTag = HostTag.ELECTED_MASTER,
+            tuple = {
+                @ATuple(
+                    dimensionValues = {GCType.Constants.OLD_GEN_VALUE, JvmSizingITConstants.CMS_COLLECTOR},
+                    sum = 10.0, avg = 10.0, max = 10.0, min = 10.0
+                )
+            }
+        )
+    }
+)
+public class HeapSizeIncreaseHighThresholdTest {
+
+  private static final int S_TO_MS = 1000;
+  public static final int SLEEP_DURATION_IN_S = 190;
+
+  @Test
+  @AExpect(
+      what = Type.DB_QUERY,
+      on = HostTag.ELECTED_MASTER,
+      validator = HeapSizeIncreaseNonBreachingValidator.class,
+      forRca = PersistedAction.class,
+      timeoutSeconds = 240
+  )
+  @AErrorPatternIgnored(
+      pattern = "AggregateMetric:gather()",
+      reason = "CPU metrics are expected to be missing in this integ test")
+  @AErrorPatternIgnored(
+      pattern = "Metric:gather()",
+      reason = "Metrics are expected to be missing in this integ test")
+  @AErrorPatternIgnored(
+      pattern = "NodeConfigCacheReaderUtil",
+      reason = "Node Config Cache are expected to be missing in this integ test.")
+  @AErrorPatternIgnored(
+      pattern = "SubscribeResponseHandler:onError()",
+      reason = "A unit test expressly calls SubscribeResponseHandler#onError, which writes an error log")
+  @AErrorPatternIgnored(
+      pattern = "SQLParsingUtil:readDataFromSqlResult()",
+      reason = "Old gen metrics is expected to be missing in this integ test.")
+  @AErrorPatternIgnored(
+      pattern = "HighHeapUsageOldGenRca:operate()",
+      reason = "Old gen rca is expected to be missing in this integ test.")
+  @AErrorPatternIgnored(
+      pattern = "ModifyCacheMaxSizeAction:build()",
+      reason = "Node config cache is expected to be missing during shutdown")
+  @AErrorPatternIgnored(
+      pattern = "NodeConfigCollector:collectAndPublishMetric()",
+      reason = "Shard request cache metrics is expected to be missing")
+  @AErrorPatternIgnored(
+      pattern = "CacheUtil:getCacheMaxSize()",
+      reason = "Shard request cache metrics is expected to be missing.")
+  @AErrorPatternIgnored(
+      pattern = "HighHeapUsageYoungGenRca:operate()",
+      reason = "YoungGen metrics is expected to be missing."
+  )
+  @AErrorPatternIgnored(
+      pattern = "PersistableSlidingWindow:<init>()",
+      reason = "Persistence base path can be null for integration test."
+  )
+  @AErrorPatternIgnored(
+      pattern = "OldGenRca:getMaxHeapSizeOrDefault()",
+      reason = "YoungGen metrics is expected to be missing.")
+  @AErrorPatternIgnored(
+      pattern = "BucketizedSlidingWindow:next()",
+      reason = "Since the persistence path can be null for integration test, calls to next() is "
+          + "expected to fail")
+  public void testDataNodeThresholdNotBreached() {
+    // We know that it takes at most 180 seconds(in the case of a multinode cluster) to fire the
```

Comment:
Does this 180s constraint apply to all ITs or this specific IT ? Do we want to move it into a base line IT test case class to share it across all ITs ?

Reply:
It applies only to this particular IT, because it takes three minutes for the unhealthiness to manifest as an action for this decider, but I'm not sure how it works for the other deciders.